### PR TITLE
Feature fix monitoring ids

### LINKED ISSUE #347: Review monitoring IDs

### PR DESCRIPTION
**Pull request by idelcano**
closes #347

### CODE REVIEW

**Comment by adrianq** [comment]
@idelcano  Extract methods common to all variant in SurveyMonitor. Move them to a SuperClass or util class, to be analysed.

**Comment by idelcano** [comment]
@adrianq  its done

File: app/src/main/java/org/eyeseetea/malariacare/monitor/utils/BaseSurveyMonitor.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ * Created by idelcano on 03/08/2016.
+ *
+ * BaseSurveyMonitor contains the common SurveyMonitor methods to each variant
+ */
+public class BaseSurveyMonitor {
+
+    /**
+     * Reference to inner survey
+     */
+    final Survey survey;
```
Comment (adrianq) [inline app/src/main/java/org/eyeseetea/malariacare/monitor/utils/BaseSurveyMonitor.java]: @idelcano  should this be private? Don't you have a getter? Is there any need to make it final?

File: app/src/main/java/org/eyeseetea/malariacare/monitor/utils/BaseSurveyMonitor.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        }
+        //No matches -> null
+        return null;
+    }
+
+    /**
+     * Returns the surveys that have been sent during the last 6 months in order to create monitor stats on top of them.
+     * @return
+     */
+    public static List<Survey> findSentSurveysForMonitor() {
```
Comment (adrianq) [inline app/src/main/java/org/eyeseetea/malariacare/monitor/utils/BaseSurveyMonitor.java]: @idelcano Don't like too much this idea of having a static method in the base class. As far as I know this method is only used from one place (it is not variant specifc). Can we extract the code from here and put it [here](https://github.com/EyeSeeTea/pictureapp/pull/351/commits/3dec5abdc7c6a26f0bf7d9b936e69f6278843c4d#diff-30d597828ea2242f465cc0c11831d0c5R99)? In general it is better to have static method in utils class

File: app/src/cambodia/java/org/eyeseetea/malariacare/monitor/utils/SurveyMonitor.java [hunk 3]
```diff
@@ -21,39 +21,39 @@
...
+public class SurveyMonitor extends BaseSurveyMonitor{
 
+    public SurveyMonitor(Survey survey){
+        super(survey);
+    }
     /**
      * Id of first question (positive, negative, not tested)
      */
-    private final static Long ID_QUESTION_RDT=1l;
+    protected final static Long ID_QUESTION_RDT=1l;
```
Comment (adrianq) [inline app/src/cambodia/java/org/eyeseetea/malariacare/monitor/utils/SurveyMonitor.java]: @idelcano  something minor but do we have to specify the field as 'protected'? I though default scope is almost the same as protected...

**Comment by adrianq** [comment]
@idelcano In general, it looks good to me but if you can have a quick look to some comments I threw into the PR and let me know you thoughts or change it accordingly. Thanks!

File: app/src/main/java/org/eyeseetea/malariacare/monitor/utils/BaseSurveyMonitor.java (hunk 1 above)
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/monitor/utils/BaseSurveyMonitor.java]: A getter method is not necessary here, but i don't know why the survey is final, I change it to private.

**Comment by idelcano** [comment]
@adrianq  it's done!, thanks for the comments!

**Comment by adrianq** [comment]
@idelcano just merged it. Just change some minor stuff: remove unneeded function, move some constants to the base class and some cleanup. Have a look at my latest commit!

**Pull request merged**